tests/unittest.py

```diff
@@ -585,15 +587,25 @@ def setup_test_homeserver(self, *args: Any, **kwargs: Any) -> HomeServer:
         else:
             config = kwargs["config"]
 
+        # The server name can be specified using either the `name` argument or a config
+        # override. The `name` argument takes precedence over any config overrides.
+        if name is not None:
+            config["server_name"] = name
+
         # Parse the config from a config dict into a HomeServerConfig
         config_obj = make_homeserver_config_obj(config)
         kwargs["config"] = config_obj
 
+        # The server name in the config is now `name`, if provided, or the `server_name`
+        # from a config override, or the default of "test". Whichever it is, we
+        # construct a homeserver with a matching name.
+        kwargs["name"] = config_obj.server.server_name
+
```

Comment:
We're inconsistent about whether we use `hs.hostname` or `hs.config.server.server_name` in Synapse. Prior to this change, they could contain entirely different values during the tests.

Reply:
Having to shove this *back* into `kwargs` is a bit annoying, but oh well.

Reply:
We could also copy this back into `name` and pass `name` to `setup_test_homeserver`. It's unpleasant either way.

Reply:
Yeah, all the config code around tests is annoying. I think this is fine.